flink-tests/src/test/java/org/apache/flink/test/checkpointing/SavepointITCase.java

```diff
@@ -600,6 +616,171 @@ public void testSubmitWithUnknownSavepointPath() throws Exception {
         }
     }
 
+    @Test
+    public void testStopWithSavepointFailingInSnapshotCreation() throws Exception {
+        testStopWithFailingSourceInOnePipeline(
+                new SnapshotFailingInfiniteTestSource(),
+                folder.newFolder(),
+                // two restarts expected:
+                // 1. task failure restart
+                // 2. job failover triggered by the CheckpointFailureManager
+                2,
+                assertInSnapshotCreationFailure());
+    }
+
+    @Test
+    public void testStopWithSavepointFailingAfterSnapshotCreation() throws Exception {
```

Comment:
This test fails with a timeout if I run it in IDE without other tests (while others don't; and also if I run it `SavepointITCase` it succeeds).

Reply:
Good catch. I missed running them individually. This test failure would be fixed if we apply the change you suggested above to run the error handling in the `mainThreadExecutor`.

Reply:
This is fixed after letting the error handling run in the `mainThreadExecutor` again.